### PR TITLE
chore(wasm): testing Kong/ngx_wasm_module#546

### PR DESCRIPTION
**Pull request by flrgh**
Created to track testing of https://github.com/Kong/ngx_wasm_module/pull/546

Do not merge me.

KAG-4671

### CODE REVIEW

**Comment by flrgh** [comment]
Testing locally, my initial findings are there appears to be something going wrong with sending a response from a filter. [This test](https://github.com/Kong/kong/blob/074a61f31c6b4c60f4b2705b7f31a768b725c579/spec/02-integration/20-wasm/04-proxy-wasm_spec.lua#L324-L342) and others like it are failing with a timeout as the client never receive a response.

In `master` I have the following error.log output for the test:

```
2024/05/28 11:34:47 [debug] 3286943#0: *2 [proxy-wasm] "tests" filter reusing instance (ictx: 000000000185D948, store: 000000000123BD28)
2024/05/28 11:34:47 [info] 3286943#0: *2 [proxy-wasm]["tests" #1] [proxy-wasm root] create_http_context (id: #1), client: 127.0.0.1, server: kong, request: "GET /single/status/200 HTTP/1.1", host: "0.0.0.0:9000", request_id: "b395de695a18a75825b491613c682167"
2024/05/28 11:34:47 [debug] 3286943#0: *2 [proxy-wasm]["tests" #1] filter 1/1 resuming "on_request_headers" step in "rewrite" phase
2024/05/28 11:34:47 [info] 3286943#0: *2 [proxy-wasm]["tests" #1] [proxy-wasm http] on_request_headers (5 headers, eof: false), client: 127.0.0.1, server: kong, request: "GET /single/status/200 HTTP/1.1", host: "0.0.0.0:9000", request_id: "b395de695a18a75825b491613c682167"
2024/05/28 11:34:47 [info] 3286943#0: *2 [proxy-wasm]["tests" #1] [proxy-wasm] testing in "RequestHeaders", client: 127.0.0.1, server: kong, request: "GET /single/status/200 HTTP/1.1", host: "0.0.0.0:9000", request_id: "b395de695a18a75825b491613c682167"
2024/05/28 11:34:47 [info] 3286943#0: *2 [proxy-wasm]["tests" #1] [proxy-wasm] kong.client.protocol: ""http"", client: 127.0.0.1, server: kong, request: "GET /single/status/200 HTTP/1.1", host: "0.0.0.0:9000", request_id: "b395de695a18a75825b491613c682167"
2024/05/28 11:34:47 [debug] 3286943#0: *2 [proxy-wasm] setting next action: pwctx->action = "DONE" (pwctx: 0000000005AED380)
2024/05/28 11:34:47 [debug] 3286943#0: *2 [proxy-wasm] setting next action: pwctx->action = "CONTINUE" (pwctx: 0000000005AED380)
2024/05/28 11:34:47 [debug] 3286943#0: *2 [proxy-wasm] resetting filter chain: pwctx->exec_index 0 to 0 (pwctx: 0000000005AED380)
2024/05/28 11:34:47 [debug] 3286943#0: *2 [proxy-wasm]["tests" #1] filter 1/1 resuming "on_response_headers" step in "header_filter" phase
2024/05/28 11:34:47 [info] 3286943#0: *2 [proxy-wasm]["tests" #1] [proxy-wasm http] on_response_headers (8 headers, eof: false), client: 127.0.0.1, server: kong, request: "GET /single/status/200 HTTP/1.1", host: "0.0.0.0:9000", request_id: "b395de695a18a75825b491613c682167"
2024/05/28 11:34:47 [debug] 3286943#0: *2 [proxy-wasm]["tests" #1] filter 1/1 resuming "on_response_body" step in "body_filter" phase
2024/05/28 11:34:47 [info] 3286943#0: *2 [proxy-wasm]["tests" #1] [proxy-wasm http] on_response_body (5 bytes, eof true), client: 127.0.0.1, server: kong, request: "GET /single/status/200 HTTP/1.1", host: "0.0.0.0:9000", request_id: "b395de695a18a75825b491613c682167"
2024/05/28 11:34:47 [debug] 3286943#0: *2 [proxy-wasm]["tests" #1] filter 1/1 resuming "on_log" step in "log" phase
2024/05/28 11:34:47 [info] 3286943#0: *2 [proxy-wasm]["tests" #1] [proxy-wasm http] on_log while logging request, client: 127.0.0.1, server: kong, request: "GET /single/status/200 HTTP/1.1", host: "0.0.0.0:9000", request_id: "b395de695a18a75825b491613c682167"
2024/05/28 11:34:47 [debug] 3286943#0: *2 [proxy-wasm]["tests" #1] filter 1/1 resuming "on_done" step in "done" phase
2024/05/28 11:34:47 [debug] 3286943#0: *2 [proxy-wasm]["tests" #1] filter 1/1 finalizing context
2024/05/28 11:34:47 [debug] 3286943#0: *2 [proxy-wasm] "tests" filter freeing context #1 (1/1)
```

Whereas in `chore/wasm-test-lua-bridge-refactor` we never seem to make it to the other phase handlers:

```
2024/05/28 11:37:07 [debug] 3317361#0: *2 [proxy-wasm] "tests" filter reusing instance (ictx: 000000000573C440, store: 0000000001038918)
2024/05/28 11:37:07 [info] 3317361#0: *2 [proxy-wasm]["tests" #1] [proxy-wasm root] create_http_context (id: #1), client: 127.0.0.1, server: kong, request: "GET /single/status/200 HTTP/1.1", host: "0.0.0.0:9000", request_id: "fdf27806b2ce595919ad0177a46c7c12"
2024/05/28 11:37:07 [debug] 3317361#0: *2 [proxy-wasm]["tests" #1] filter 1/1 resuming "on_request_headers" step in "access" phase
2024/05/28 11:37:07 [info] 3317361#0: *2 [proxy-wasm]["tests" #1] [proxy-wasm http] on_request_headers (5 headers, eof: false), client: 127.0.0.1, server: kong, request: "GET /single/status/200 HTTP/1.1", host: "0.0.0.0:9000", request_id: "fdf27806b2ce595919ad0177a46c7c12"
2024/05/28 11:37:07 [info] 3317361#0: *2 [proxy-wasm]["tests" #1] [proxy-wasm] testing in "RequestHeaders", client: 127.0.0.1, server: kong, request: "GET /single/status/200 HTTP/1.1", host: "0.0.0.0:9000", request_id: "fdf27806b2ce595919ad0177a46c7c12"
2024/05/28 11:37:07 [info] 3317361#0: *2 [proxy-wasm]["tests" #1] [proxy-wasm] kong.client.protocol: ""http"", client: 127.0.0.1, server: kong, request: "GET /single/status/200 HTTP/1.1", host: "0.0.0.0:9000", request_id: "fdf27806b2ce595919ad0177a46c7c12"
2024/05/28 11:37:07 [debug] 3317361#0: *2 [proxy-wasm] setting next action: pwctx->action = "DONE" (pwctx: 00000000049F0C20)
2024/05/28 11:37:07 [debug] 3317361#0: *2 [proxy-wasm] setting next action: pwctx->action = "CONTINUE" (pwctx: 00000000049F0C20)

[no more output after this]
```

File: kong/runloop/wasm.lua [hunk 1]
```diff
@@ -945,13 +945,6 @@ function _M.attach(ctx)
...
     end
   end
-
-  jit.off(proxy_wasm.start)
-  ok, err = proxy_wasm.start()
-  if not ok then
-    log(CRIT, "failed to execute ", chain.label, " filter chain for request: ", err)
-    return kong.response.error(500)
-  end
 end
```
Comment (flrgh) [inline kong/runloop/wasm.lua:948]: Paging @zhongweiy.

From https://github.com/Kong/ngx_wasm_module/pull/546, you'll see that `proxy_wasm.start()` is removed and no longer used, so we're no longer manually disabling jit here.

When you have time, can you look at the code changes and advise if we need to disable jit anywhere else instead (perhaps `proxy_wasm.attach()`)? I'm curious if the conditions of a796ac6105b60fbd0e85c24281b9f91cf19bdeaf still apply somewhere.

Comment (zhongweiy) [reply kong/runloop/wasm.lua:948]: Once there is no `ffi call (possible to be compiled)` -> `c function` -> `lua callback function` call chain, it is free from the previous `bad callback` error.

It looks there is no such `ffi call` entry call any more in current design. If yes, I think it is OK to not do `jit.off`.

**Comment by flrgh** [comment]
Closing in favor of #13222

**Pull request closed without merging**